pkg/resources/machinecontroller/deployment.go

```diff
@@ -281,6 +285,10 @@ func getFlags(clusterDNSIP string, nodeSettings *kubermaticv1.NodeSettings, cri
 			flags = append(flags, "-node-pause-image", nodeSettings.PauseImage)
 		}
 
+		if isExternal {
+			flags = append(flags, "-node-external-cloud-provider")
+		}
+
```

Comment:
this part is not needed, if the cloud provider is external then machine deployment webhook will set appropriate kubelet flag. Look at https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/machinecontroller/webhook.go#L65-L68

Reply:
thnx for the hint
Done